lib/Component.spec.lua

```diff
@@ -207,6 +211,90 @@ return function()
 			end).to.throw()
 		end)
 
+		it("should throw when called in shouldUpdate", function()
+			local TestComponent = Component:extend("TestComponent")
+
+			function TestComponent:render()
+				return nil
+			end
+
+			function TestComponent:shouldUpdate()
+				self:setState({
+					a = 1
+				})
+			end
+
+			local testElement = Core.createElement(TestComponent)
+
+			expect(function()
+				local handle = Reconciler.reify(testElement)
+				-- Something of a hack, but...
+				handle._instance:shouldUpdate({}, {})
```

Comment:
The other thing you can do is capture the instances `setState` function from `init`, and then call that to trigger `shouldUpdate`/`willUpdate`/`didUpdate`.

It's quite a bit more verbose, but it avoids leaking internals into tests! I don't know if it's actually better; what do you think?

Reply:
It's a lot better, because then I'm not depending on the format of the instance handles (oh no, we changed the reified handle format, our tests for setState broke!). I'll make the change.